transformer_engine/common/fused_attn/fused_attn_fp16_bf16_max_seqlen_512.cu

```diff
@@ -1328,20 +1383,26 @@ void fused_attn_max_512_fwd_kvpacked(size_t batch, size_t q_max_seqlen, size_t k
     void *devQCuSeqlen = q_cu_seqlens->data.dptr;
     void *devKVCuSeqlen = kv_cu_seqlens->data.dptr;
 
-    // TODO(rewang): dropout seed
-    // void* devPtrDropoutSeed = reinterpret_cast<void *>(
-    //                 reinterpret_cast<uint64_t*>(rng_state->data.dptr));
-    // void* devPtrDropoutOffset = reinterpret_cast<void *>(
-    //                 reinterpret_cast<uint64_t*>(rng_state->data.dptr) + 1);
+    const DType rng_state_type = rng_state->data.dtype;
+    void *devPtrDropoutSeed = rng_state->data.dptr;
+    void *devPtrDropoutOffset = nullptr;  // Set according to the rng_state_type
+    if (rng_state_type == DType::kInt32) {
+        devPtrDropoutOffset =
+            static_cast<void *>(static_cast<uint32_t *>(rng_state->data.dptr) + 1);
```

Comment:
Would it be better to move this conversion to the framework side, so on the C side we always take in a uint64 pair as the seed and offset? The different types and numbers of values in the rng state are caused by the different random number generators used on the framework side.